adding the automation of BZ : 1829646 - Autotrimming of OSD map and MonDB when OSD(s) is in down state

pdhiran opened the pull request:
Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

Added the test in polarion for the same.
The test cannot be run on a newly created cluster as monDB keeps in increasing for around 1 hour with new osd mappings. Hence placed the test at the end of rados suite tests so that cluster will be aged sufficiently.

Pass logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1594878423461/

pdhiran commented:
Addressing review comments from previous PR:

is bug present in luminous too? if so please add it to luminous sanity_rados_lvm suite too
-> The fix is not back ported to luminous. Test can be added in luminous suite once that is done.

there is function to bring down/uposd in rados_utils ..can we improve that to use it here?is it compatible for this bz?
-> The command seems to be slightly different and i'm using the same method to start/stop both OSD daemons and nodes. Will modify the method later.  That would need significant code change.

can we move this to rados_utils.if possible..looks like it'll be helpful to everyone.Even i had to use this ls-tree in one of the bz
-> moved the method to rados_utils as suggested.

XDsubhash commented on tests/rados/test_bz_1829646.py:
```diff
@@ -0,0 +1,219 @@
+    :return: None
+    """
+
+    log.info(f"inside method to {task} the OSD(s) on the node : {osd_node.hostname}")
```
from next time try to avoid "inside method" string

pdhiran replied:
Should i change the string in the print here then? Sure, will keep it in mind and avoid that in future.

XDsubhash commented:
> Addressing review comments from previous PR:
> 
> is bug present in luminous too? if so please add it to luminous sanity_rados_lvm suite too
> -> The fix is not back ported to luminous. Test can be added in luminous suite once that is done.
the bz was found in 3.3 , we need to include in luminous too

pdhiran commented:
> > Addressing review comments from previous PR:
> > is bug present in luminous too? if so please add it to luminous sanity_rados_lvm suite too
> > -> The fix is not back ported to luminous. Test can be added in luminous suite once that is done.
> 
> the bz was found in 3.3 , we need to include in luminous too

3.3 BZ : Bug 1798833 was in closed state with Resolution: --- → NEXTRELEASE. Hence had not included the test in luminous suite. We can add it to suite if the fix will be added to luminous.

XDsubhash commented:
as per dev , it wont be backported to 3.x.

XDsubhash commented:
@pdhiran Pl mark customer scenario checkbox in polarion and make sure polarion tc is approved and automated
